### ISSUE TITLE
Problems with collapsible blocks 

### ISSUE THREAD

**Issue by ghost** [ggrossetie/asciidoctor-web-pdf#241]
I use a collapsible block in a test document.
 ```
[%collapsible]
====
Example block turns into collapsible summary/details.
====
```

The collapsible is not automatically expanded in the PDF generated. It works if I write it
```
[%collapsible%open]
```
I think, collapsibles should always be expanded when producing PDF's. Or at least there should be a CLI-option to expand all collapsible blocks.

If the collapsible is expanded with %open, the text 'Details' gets displayed on every page that contains part of the collapsible. That is a) not necessary and b) does not display the collapsible title that has been set but instead the string 'Details' which is the default text. 

I attach a PDF that demonstrates that. It has a little more content than the sample code above.

[sample.pdf](https://github.com/Mogztter/asciidoctor-pdf.js/files/4695148/sample.pdf)

**Comment by Mogztter** [comment]
That's a good point!

To be clear, the sample.pdf file attached was generated from Asciidoctor Web PDF (I really need to rename this project I keep saying Asciidoctor Web PDF but the project is named Asciidoctor.js PDF... anyway :zany_face:) or Asciidoctor Ruby PDF?

**Comment by henriette-einstein** [comment]
Yes, the PDF has been generated with Asciidoctor Web PDF. Asciidoctor-pdf --version returns

```
Asciidoctor PDF 1.0.0-alpha.7 using Asciidoctor.js 2.2.0 (Asciidoctor 2.0.10) [https://asciidoctor.org]
Runtime Environment (node v14.0.0 on darwin)
CLI version 3.2.0
```

**Comment by Mogztter** [comment]
Thanks I will have a look :eyes: 
I was asking because I wonder how Asciidoctor Ruby PDF (the "reference implementation") is solving this issue.

**Comment by Mogztter** [comment]
I can confirm that collapsible are "expanded" with Asciidoctor PDF.

[collapsible.pdf](https://github.com/Mogztter/asciidoctor-web-pdf/files/5135538/collapsible.pdf)

I don't think we can "open" the collapsible block using CSS.
To fix this issue, we could:
* Use JavaScript to set the `open` attribute on `<details>` elements
* Override the built-in HTML 5 converter to always add the  `open` attribute
* Use a TreeProcessor extension to add the `open` option on collapsible blocks

Any preference?

**Comment by henriette-einstein** [comment]
I'd probably use a TreeProcessor.

- I would not aus JavaScript. It adds additional complexity to the processing chain with probably additional problems.
- The HTML 5 converter does what it has to do. It produces HTML 5. That the PDF production has to fiddle with the HTML in order to expand collapsible does not belong its responsibility.

BTW: I am dealing with some personal issues to get rid of some errand cells using chemicals. Therefore I am not participating too muh at the moment. I hopefully will be back soon.

**Comment by Mogztter** [comment]
> BTW: I am dealing with some personal issues to get rid of some errand cells using chemicals. Therefore I am not participating with Asciidoctor stuff too much at the moment. I hopefully will be back soon.

I'm sorry to hear that...
Take care, I hope you will get better soon :pray: 

> I would not aus JavaScript. It adds additional complexity to the processing chain with probably additional problems.

Yes, I agree :100: 


> The HTML 5 converter does what it has to do. It produces HTML 5. That the PDF production has to fiddle with the HTML in order to expand collapsible does not belong its responsibility.

Yes, but at the same time the converter is specific and its responsibility is to produce a *printable* HTML 5 page. So I think it's reasonable to do that in the converter.

Currently, we rely a lot on the built-in HTML 5 converter from Asciidoctor but it was mainly to get a "quick" result at the early stage of the project. Now that this project is gradually becoming a full-fledged converter, we can add more logic.

**Comment by ggrossetie** [comment]
Fixed on `fix/bucket-d-remaining` (not yet merged): https://github.com/ggrossetie/asciidoctor-web-pdf/tree/fix/bucket-d-remaining
